pkg/vpcmodel/abstractVPC.go

```diff
@@ -74,17 +74,40 @@ type NodeSet interface {
 	AddressRange() *common.IPBlock
 }
 
+type RulesInFilter struct {
+	// todo: is the assumption that the set of rules will always be kept in a list a valid one?
+	RulesIndexes int   // sg/nacl index in sgList/naclList
+	Rules        []int // list of indexes if rules in the sg/nacl
+
+}
+
+type rulesInLayer struct {
+	layer string
+	rules []RulesInFilter
+}
+
+type rulesInLayers []rulesInLayer
+
+type RulesOfConnection struct {
+	ingressRules rulesInLayers
+	egressRules  rulesInLayers
+}
```

Comment:
Documentation added.
Regarding the location of the definitions -  `FilterTrafficResource interface` uses these `[]RulesInFilter` which implies `RulesInFilter` should stay here. I thought it would be better to keep all types together since they are based on each others. Do you think it is better to move `rulesInLayer, rulesInLayers` and `RulesOfConnection` to `nodesExplainability.go`?